driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -35,21 +37,23 @@
  */
 public final class DefaultDnsResolver implements DnsResolver {
 
+    private static final DnsClient DEFAULT_DNS_CLIENT;
+
+    static {
+        DEFAULT_DNS_CLIENT = StreamSupport.stream(ServiceLoader.load(DnsClientProvider.class).spliterator(), false)
```

Comment:
Simplify the code to match what's now done in `ServerAddressWithResolver`

Reply:
The changes made in this PR require `DnsClient` to be thread-safe, which is a breaking change. Previosuly, `DnsClient` was not marked thread-safe and it was not required to be threadsafe:
- A new instance of `DnsClient` was always created in the constructor of `DefaultDnsResolver` (`DnsClientProvider.create` is documented as constructing a new instance).
- A new instance of `DefaultDnsResolver` was created and used in the constructor of `ConnectionString`.
- A new instance of `DefaultDnsResolver` was created per `DefaultDnsSrvRecordMonitor`, where it was used by the single corresponding monitoring thread.

Reply:
I see you have already asked the Quarkus team in https://jira.mongodb.org/browse/JAVA-4512. I agree that we can just make the change if their implementation is thread-safe.

Reply:
It looks like it is.

Reply:
Added ThreadSafe annotation to both interfaces